content/php-api-using-jwt/index.md

```diff
@@ -0,0 +1,226 @@
+$issuedAt   = new DateTimeImmutable();
+$expire     = $issuedAt->modify('+6 minutes')->getTimestamp();      // Add 60 seconds
+$serverName = "your.domain.name";
+$username   = "username";                                           // Retrieved from filtered POST data
+$data = [
+    'iat'  => $issuedAt->getTimestamp(),         // Issued at: time when the token was generated
+    'iss'  => $serverName,                       // Issuer
+    'nbf'  => $issuedAt->getTimestamp(),         // Not before
+    'exp'  => $expire,                           // Expire
+    'userName' => $username,                     // User name
+];
+```
+Now we have all the required data in hand, so we can easily create a JWT. Here, we'll use the PHP-JWT package's encode() method. This method is quite useful in transforming our data array into a JSON object.
+
+Following the conversion to a JSON object, the encode function produces JWT headers and signs the received payload with a cryptographic combination of all the information and the given secret key.
+
+It is important to supply three arguments to the encode method in order to utilize it correctly. The first argument should be the payload information, which in our instance is the data array. Secondly, you must supply the secret key as an argument, and finally, you must define the cryptographic technique that the function should use to sign the JWT.
+
+To obtain and return the JWT, you'll have to use the echo method above the encode method, as shown below.
+
+```php
+<?php
+    // Encode the array to a JWT string.
+    echo JWT::encode(
+        $data,
+        $secretKey,
+        'HS512'
+    );
+}
+```
+Now that we have obtained the JWT token, we can transfer it to the client-side and save it using any web programming language of our choice. We'll start with a short JS demonstration of the route ahead of us.
+
+To begin, when a successful form submission takes place, save the created and received JWT in client-side memory. To display some output about the JWT's success, remove the login form and merely display a button that retrieves and displays the JWT's timestamp to the user when it is clicked.
+
+Here is some sample code for the aforementioned process.
+```javascript
+const store = {};
+const loginButton = document.querySelector('#frmLogin');
+const btnGetResource = document.querySelector('#btnGetResource');
+const form = document.forms[0];
+
+// Inserts the jwt to the store object
+store.setJWT = function (data) {
+  this.JWT = data;
+};
+
+loginButton.addEventListener('submit', async (e) => {
+  e.preventDefault();
+
+  const res = await fetch('/authenticate.php', {
+    method: 'POST',
+    headers: {
+      'Content-type': 'application/x-www-form-urlencoded; charset=UTF-8'
+    },
+    body: JSON.stringify({
+      username: form.inputEmail.value,
+      password: form.inputPassword.value
+    })
+  });
+
+  if (res.status >= 200 && res.status <= 299) {
+    const jwt = await res.text();
```

Comment:
This code block is exactly matching with this source: PHP Authorization with JWT (JSON Web Tokens) - SitePoint. https://www.sitepoint.com/php-authorization-jwt-json-web-tokens/

Can you please try to make it any more unique by modifying the code without disturbing what the code tries to achieve in terms of logic and flow?

Reply:
Done